src/modules/vtol_att_control/standard.cpp

```diff
@@ -277,39 +232,36 @@ void Standard::update_vtol_state()
 void Standard::update_transition_state()
 {
 	float mc_weight = 1.0f;
+	float time_since_trans_start = (float)(hrt_absolute_time() - _vtol_schedule.transition_start) * 1e-6f;
 
 	VtolType::update_transition_state();
 
 	// copy virtual attitude setpoint to real attitude setpoint
 	memcpy(_v_att_sp, _mc_virtual_att_sp, sizeof(vehicle_attitude_setpoint_s));
 
 	if (_vtol_schedule.flight_mode == TRANSITION_TO_FW) {
-		if (_params_standard.front_trans_dur <= 0.0f) {
+		if (_params_standard.pusher_ramp_dt <= 0.0f) {
 			// just set the final target throttle value
-			_pusher_throttle = _params_standard.pusher_trans;
+			_pusher_throttle = _params->front_trans_throttle;
 
-		} else if (_pusher_throttle <= _params_standard.pusher_trans) {
+		} else if (_pusher_throttle <= _params->front_trans_throttle) {
 			// ramp up throttle to the target throttle value
-			_pusher_throttle = _params_standard.pusher_trans *
-					   (float)hrt_elapsed_time(&_vtol_schedule.transition_start) / (_params_standard.front_trans_dur * 1000000.0f);
+			_pusher_throttle = _params->front_trans_throttle * time_since_trans_start / _params_standard.pusher_ramp_dt;
 		}
 
 		// do blending of mc and fw controls if a blending airspeed has been provided and the minimum transition time has passed
 		if (_airspeed_trans_blend_margin > 0.0f &&
-		    _airspeed->indicated_airspeed_m_s >= _params_standard.airspeed_blend &&
-		    (float)hrt_elapsed_time(&_vtol_schedule.transition_start) > (_params_standard.front_trans_time_min * 1000000.0f)
-		   ) {
-			mc_weight = 1.0f - fabsf(_airspeed->indicated_airspeed_m_s - _params_standard.airspeed_blend) /
+		    _airspeed->indicated_airspeed_m_s >= _params->airspeed_blend &&
+		    time_since_trans_start > _params->front_trans_time_min) {
+			mc_weight = 1.0f - fabsf(_airspeed->indicated_airspeed_m_s - _params->airspeed_blend) /
 				    _airspeed_trans_blend_margin;
 			// time based blending when no airspeed sensor is set
 
-		} else if (_params_standard.airspeed_disabled &&
-			   hrt_elapsed_time(&_vtol_schedule.transition_start) < (_params_standard.front_trans_time_min * 1e6f) &&
-			   hrt_elapsed_time(&_vtol_schedule.transition_start) > ((_params_standard.front_trans_time_min / 2.0f) * 1e6f)
-			  ) {
-			mc_weight = 1.0f - ((float)(hrt_elapsed_time(&_vtol_schedule.transition_start) - ((
-							    _params_standard.front_trans_time_min / 2.0f) * 1000000.0f)) /
-					    ((_params_standard.front_trans_time_min / 2.0f) * 1000000.0f));
+		} else if (_params->airspeed_disabled &&
+			   time_since_trans_start < _params->front_trans_time_min &&
+			   time_since_trans_start > _params->front_trans_time_min / 2.0f) {
+			mc_weight = 1.0f - ((time_since_trans_start -  _params->front_trans_time_min / 2.0f) /
```

Comment:
I believe this simplifies to
`float weight = 1.0f - (time_since_trans_start / _params->front_trans_time_min);`

Reply:
@dagar Not really, I think the intent is to start the weight from zero when time_since_trans_start is half of front_trans_time_min.
By the way, I mostly wanted to keep this PR as pure refactoring and not change any functional stuff.

Reply:
I'm talking about keeping it the same functionally, but in a somewhat comprehensible form.

    const float weight = 1.0f - (time_since_trans_start / _params->front_trans_time_min);
    _mc_throttle_weight = math::constrain(2.0f * weight, 0.0f, 1.0f);